csv module typing of dialect parameters is not consistently correct

This code

    import csv

    d = csv.excel
    d = csv.Sniffer().sniff("a")

    with open('eggs.csv', newline='') as csvfile:
        spamreader = csv.DictReader(csvfile, dialect=csv.excel)

gives me two errors:

    testmypy.py:4: error: Incompatible types in assignment (expression has type "Dialect", variable has type "Type[excel]")
    testmypy.py:7: error: Argument "dialect" to "DictReader" has incompatible type "Type[excel]"; expected "Union[str, Dialect]"

The problem seems to be that some bits of the stub for `csv` use type `Dialect` when they should be using `type[Dialect]`.  The return of `sniff` is one and so is the type of the `dialect` argument to the `DictReader` constructor.  There may be more ...

I'm using Python 3.7.3 and mypy 0.701 but I think the problems are general.

--- a/stdlib/2and3/csv.pyi
+++ b/stdlib/2and3/csv.pyi
@@ -1,6 +1,6 @@
 from collections import OrderedDict
 import sys
-from typing import Any, Dict, Iterable, Iterator, List, Mapping, Optional, Sequence, Union
+from typing import Any, Dict, Iterable, Iterator, List, Mapping, Optional, Sequence, Type, Union
 
 from _csv import (_reader,
                   _writer,
@@ -18,7 +18,7 @@ from _csv import (_reader,
                   Error as Error,
                   )
 
-_Dialect = Union[str, Dialect]
+_Dialect = Union[str, Dialect, Type[Dialect]]
 _DictRow = Mapping[str, Any]
 
 class Dialect(object):
@@ -89,5 +89,5 @@ class DictWriter(object):
 class Sniffer(object):
     preferred: List[str]
     def __init__(self) -> None: ...
-    def sniff(self, sample: str, delimiters: Optional[str] = ...) -> Dialect: ...
+    def sniff(self, sample: str, delimiters: Optional[str] = ...) -> Type[Dialect]: ...
     def has_header(self, sample: str) -> bool: ...

